sklearn/impute/_knn.py

```diff
@@ -50,10 +50,11 @@ class KNNImputer(_BaseImputer):
           array of distances, and returns an array of the same shape
           containing the weights.
 
-    metric : {'nan_euclidean'} or callable, default='nan_euclidean'
+    metric : {'nan_euclidean', 'nan_manhattan'} or callable, default='nan_euclidean'
```

Comment:
Since a new support is added to `KNNImputer`, I think you need to extend `sklearn/impute/tests/test_knn.py` on tests checking valid imputations to add support for `nan_manhattan`.